lib/file-source/src/file_server.rs

```diff
@@ -248,6 +283,30 @@ impl FileServer {
             }
         }
     }
+
+    fn watch_new_file(
+        &self,
+        path: PathBuf,
+        file_id: FileFingerprint,
+        fp_map: &mut IndexMap<FileFingerprint, FileWatcher>,
+        checkpointer: &Checkpointer,
+        read_from_beginning: bool,
+    ) {
+        let file_position = if read_from_beginning {
+            0
+        } else {
+            checkpointer.get_checkpoint(file_id).unwrap_or(0)
+        };
+        if let Ok(mut watcher) = FileWatcher::new(path, file_position, self.ignore_before) {
```

Comment:
should probably print an error here?

Reply:
This was just moved, but yeah good idea. I'll add it in a followup.